Enable slowly-changing, important diagnostics for connected devices by default

lanthaler opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Enable slowly-changing, important diagnostics for connected devices by default. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: n/a
- This PR is related to issue: n/a
- Link to documentation pull request: n/a

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

lanthaler committed:
Enable slowly-changing, important diagnostics for connected devices by default

joostlek commented on homeassistant/components/netgear/sensor.py:
```diff
@@ -47,28 +47,33 @@
         key="type",
         translation_key="link_type",
         entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=True,
```
```suggestion
```

joostlek commented on homeassistant/components/netgear/sensor.py:
```diff
@@ -47,28 +47,33 @@
         key="ssid",
         translation_key="ssid",
         entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=True,
```
```suggestion
```

joostlek reviewed (dismissed):
This is default. I will leave the rest to the codeowners

lanthaler commented:
Thanks @joostlek, removed the explicit default values

joostlek commented:
I think its also good to check what other network integrations do like unifi, because I don't mind to enable some, but like you say, we should not overdo it

lanthaler commented:
That's a valid concern. My use case is to treat devices on guest network differently than the main network but I can't do so without knowing the SSID and I don't know about all devices upfront.

I see a couple of options to move forward:
1) keep these sensors enabled by default but hide them (`entity_registry_visible_default`)
2) make it configurable which sensors should be enabled by default
3) remove the sensor and store this in `extra_state_attributes` in the device tracker entity instead (breaking change unless the information is stored twice and sensors are kept)
4) extend the `state_attributes` of device tracker's ScannerEntity

Any other options I forgot? WDYT about moving forward with 1)?

starkillerOG commented:
I think your use case makes sence and that the SSID can indeed be usefull.
Lets enable SSID by default but keep the link_type disabled.

lanthaler commented:
OK, I changed it so that only SSID is enabled by default and squashed and rebased everything onto dev.

I also removed the change to reset sensor values to unknown when the device disconnects as that might warrant some more discussion. I'll send a separate PR for that in a minute.
